### ISSUE TITLE
Extend SimpleFormIterator to enable/disable rendering of index

### ISSUE THREAD

**Issue by pefi1011** [marmelab/react-admin#4905]
**Is your feature request related to a problem? Please describe.**

The problem is UX-based - the index of an array is not always an useful/required/needed information

```
<ArrayInput >
          <SimpleFormIterator>
         ...
          </SimpleFormIterator>
   </ArrayInput>
```

**Describe the solution you'd like**

Conditional rendering of [typography element](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/form/SimpleFormIterator.js#L141)
```
                            <Typography
                                variant="body1"
                                className={classes.index}
                            >
                                {index + 1}
                            </Typography>
```
 

It would be in the same fashion like `disableRemove` and  `disableAdd`
```
<ArrayInput >
          <SimpleFormIterator disableIndex >
         ...
          </SimpleFormIterator>
   </ArrayInput>
```

**Describe alternatives you've considered**
As far as I understood, to disable rendering of Index I would need to implement my own SimpleFormIterator iterator

**Comment by pefi1011** [comment]
Here is the PR: https://github.com/marmelab/react-admin/pull/4908

Any feedback will be highly appreciated

**Comment by fzaninotto** [comment]
Fixed by #6305
